Tests/SwiftMailer/MailjetTransportv31Test.php

```diff
@@ -0,0 +1,336 @@
+    }
+
+    public function testSendTextEmailWithTemplateId() {
+        $transport = $this->createTransport();
+        $message = new \Swift_Message('Test Subject', 'Foo bar');
+        $message
+                ->addTo('to@example.com', 'To Name')
+                ->addFrom('from@example.com', 'From Name')
+        ;
+        $message->setBody("Hello world!");
+        $message->getHeaders()->addTextHeader('X-MJ-TemplateID', 'azertyuiop');
+        $mailjetMessage = $transport->messageFormat->getMailjetMessage($message)['Messages'][0];
+        $result = $transport->send($message);
+        $this->assertEquals('azertyuiop', $mailjetMessage['TemplateID']);
+        $this->assertMessageSendable($message);
+    }
+
+    public function testSendEmailWithAllCustomHeaders() {
+        $transport = $this->createTransport();
+        $message = new \Swift_Message('Test Subject', 'Foo bar');
+        $message
+                ->addTo('to@example.com', 'To Name')
+                ->addFrom('from@example.com', 'From Name')
+        ;
+        $message->setBody("Hello world!");
+        $message->getHeaders()->addTextHeader('X-MJ-TemplateID', 'azertyuiop');
+        $message->getHeaders()->addTextHeader('X-MJ-TemplateLanguage', true);
+        $message->getHeaders()->addTextHeader('X-MJ-TemplateErrorReporting', 'air-traffic-control@mailjet.com');
+        $message->getHeaders()->addTextHeader('X-MJ-TemplateErrorDeliver', 'deliver');
+        $message->getHeaders()->addTextHeader('X-Mailjet-Prio', 3);
+        $message->getHeaders()->addTextHeader('X-Mailjet-Campaign', 'azertyuiop');
+        $message->getHeaders()->addTextHeader('X-Mailjet-DeduplicateCampaign', false);
+        $message->getHeaders()->addTextHeader('X-Mailjet-TrackOpen', 1);
+        $message->getHeaders()->addTextHeader('X-Mailjet-TrackClick', 2);
+        $message->getHeaders()->addTextHeader('X-MJ-CustomID', 'PassengerEticket1234');
+        $message->getHeaders()->addTextHeader('X-MJ-EventPayLoad', 'Eticket,1234,row,15,seat,B');
+        $message->getHeaders()->addTextHeader('X-MJ-Vars', array('today' => 'monday'));
+        $message->getHeaders()->addTextHeader('X-MyCustomHeader', 'CustomHeader');
+
+        $mailjetMessage = $transport->messageFormat->getMailjetMessage($message)['Messages'][0];
+        $result = $transport->send($message);
+
+        $this->assertEquals('azertyuiop', $mailjetMessage['TemplateID']);
+        $this->assertEquals(true, $mailjetMessage['TemplateLanguage']);
+        $this->assertEquals('air-traffic-control@mailjet.com', $mailjetMessage['TemplateErrorReporting']);
+        $this->assertEquals('deliver', $mailjetMessage['TemplateErrorDeliver']);
+        $this->assertEquals(3, $mailjetMessage['Priority']);
+        $this->assertEquals('azertyuiop', $mailjetMessage['CustomCampaign']);
+        $this->assertEquals(false, $mailjetMessage['DeduplicateCampaign']);
+        $this->assertEquals(1, $mailjetMessage['TrackOpens']);
```

Comment:
It's a string in Send API v3.1, see https://dev.mailjet.com/beta/#send-api-json-properties